PetrolEngine.cpp

```diff
@@ -15,6 +16,27 @@ void PetrolEngine::changeGear(int gear)
     // TODO: Add checking if gear is between -1 and gears_
     // -1 is for REAR
     // 0 is for NEUTRAL
-    currentGear_ = gear;
+    
     std::cout << __FUNCTION__ << std::endl;
+        if (gear >= -1 && gear <= gears_) 
+        {
+                if (currentGear_ == -1 && gear > 1)
+                {   
+                    throw std::logic_error("Can't change gear from rear to higher than 1");
+                }
+
+                if (currentGear_ > 1 && gear == -1) 
+                {
+                    throw std::logic_error("Can't change gear from higher than one to rear");
+                }
+                
+                else {currentGear_ = gear;}
+        }
+        else {throw std::out_of_range("Invalid gear!");}
```

Comment:
Prawie o to chodziło :)
Tutaj uzywacie wyjątków z biblioteki standardowej, a mieliście utworzyć własną klasę wyjątku, która by dziedziczyła np po tym std::out_of_range i byłaby tutaj rzucana.